### ISSUE TITLE
Bogus load error after valid load failure due to missing building type

### ISSUE THREAD

**Issue by tothxa** [widelands/widelands#4861]
**Describe the bug**
It is not possible to load another game or start a new one after trying to load a game with outdated building name

**To Reproduce**
1. Load an old savegame with cassava root plantation in a version after it was renamed to cassava plantation ([example attached](https://github.com/widelands/widelands/files/6489009/cassava_root_save.zip))
2. The load will fail due to missing building type with an appropriate error message
3. Try to start a new game
4. Loading will fail with message: "Unknown immovable: 'berry_bush_blueberry_small'" (seems to be the same all the time)

**Expected behavior**
Be able to load or start another game after load failure

**Screenshots**
![load_failure_1](https://user-images.githubusercontent.com/80712255/118393254-727db280-b63e-11eb-9492-ab2285d6f36b.png)
![load_failure_2](https://user-images.githubusercontent.com/80712255/118393255-73164900-b63e-11eb-9616-a1615d34d1e1.png)

**Version:**
 - OS: antix / Debian Buster + backports
 - Widelands Version: commit 857dd74595c6da8885fcfd7d6f5f32037f2c3a73

**Additional context**
[stdout](https://github.com/widelands/widelands/files/6489023/load_failure_stdout.txt)

**Comment by Noordfrees** [comment]
The first one should not happen at all since we have compatibility code for the renamed units. Does the savegame still fail to load after recompiling with current master?

The second message seems related to #4602

… edit: yes I see, this should not happen. Will investigate soonish…

**Comment by tothxa** [comment]
This *IS* current master. It happens with flathub's newest build, as well as with previous local native and flatpak builds.

I first experienced it soon after the rename, but I just passed over it then. I only connected the second error with the first one now, and now I tested that none of these versions can start a game after failing to load an old amazons game.

The installed flatpak uses its own config without locale, while the native build and the locally built flatpak's test run uses ~/.widelands/config with locale hu.
